GVFS/GVFS.Common/FileSystem/PhysicalFileSystem.cs

```diff
@@ -177,6 +177,9 @@ public bool TryWriteTempFileAndRename(string destinationPath, string contents, o
             handledException = null;
             string tempFilePath = destinationPath + ".temp";
 
+            string parentPath = Path.GetDirectoryName(tempFilePath);

```

Comment:
@jeschu1: this is to fix problems when the parent folder doesn't exist. It seems like a good place to put it, since we generally expect the folder to exist before we try to create a file inside. It doesn't recurse to higher levels, though.

Reply:
> It doesn't recurse to higher levels, though.

FYI - `Directory.CreateDirectory` will create all missing folders in the specified path (even those at higher levels)